File: Printpress.Application/Reports/DTO/CashTreasuryReportDto.cs
namespace Printpress.Application;

public class CashTreasuryReportDto
{
    public DateTime? DateFrom { get; set; }
    public DateTime? DateTo { get; set; }
    public decimal TotalStoredBalance { get; set; }
    public List<CashTreasuryAccountDto> Accounts { get; set; } = [];
    public List<CashTreasuryMovementDto> LargestIn { get; set; } = [];
    public List<CashTreasuryMovementDto> LargestOut { get; set; } = [];
    public List<CashTransferRegisterRowDto> Transfers { get; set; } = [];
}

public class CashTreasuryAccountDto
{
    public Guid CashAccountId { get; set; }
    public string CashAccountName { get; set; }
    public string AccountType { get; set; }
    public decimal StoredBalance { get; set; }
}

public class CashTreasuryMovementDto
{
    public Guid Id { get; set; }
    public DateTime TransactionDate { get; set; }
    public string CashAccountName { get; set; }
    public decimal Amount { get; set; }
    public string Category { get; set; }
    public string Description { get; set; }
}

public class CashTransferRegisterRowDto
{
    public Guid TransferId { get; set; }
    public DateTime TransactionDate { get; set; }
    public decimal Amount { get; set; }
    public string FromAccountName { get; set; }
    public string ToAccountName { get; set; }
    public string Description { get; set; }
    public bool IsComplete { get; set; }
}

File: Printpress.UI/Printpress_Angular/src/app/features/reports/models/cash-treasury-report.dto.ts
export interface CashTreasuryReportDto {
  dateFrom: string | null;
  dateTo: string | null;
  totalStoredBalance: number;
  accounts: CashTreasuryAccountDto[];
  largestIn: CashTreasuryMovementDto[];
  largestOut: CashTreasuryMovementDto[];
  transfers: CashTransferRegisterRowDto[];
}

export interface CashTreasuryAccountDto {
  cashAccountId: string;
  cashAccountName: string;
  accountType: string;
  storedBalance: number;
}

export interface CashTreasuryMovementDto {
  id: string;
  transactionDate: string;
  cashAccountName: string;
  amount: number;
  category: string;
  description: string;
  referenceLabel?: string | null;
  referenceRoute?: string | null;
}

export interface CashTransferRegisterRowDto {
  transferId: string;
  transactionDate: string;
  amount: number;
  fromAccountName: string;
  fromAccountId?: string | null;
  toAccountName: string;
  toAccountId?: string | null;
  description: string;
  isComplete: boolean;
}
